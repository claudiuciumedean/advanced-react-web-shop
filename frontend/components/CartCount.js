import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TransitionGroup, CSSTransition } from "react-transition-group";

const Dot = styled.div`
    min-width:3rem; margin-left:1rem; padding:.5rem; border-radius:50%; font-weight:100; font-feature-settings:"enum"; font-variant-numeric:tabular-nums; line-height:2rem; color:${props => props.theme.white}; background:${props => props.theme.red};
`;

const AnimationStyles = styled.span`
    position:relative;
    .count { display:block; position:relative; backface-visibility:hidden; transition:transform .4s; }

    .count-enter { transform:rotateX(.5turn); }
    .count-enter-active { transform:rotateX(0); }

    .count-exit { position:absolute; top:0; transform:rotateX(0); }
    .count-exit-active { transform:rotateX(.5turn); }
`;

const CartCount = ({ count }) => (
    <AnimationStyles>
        <TransitionGroup>
            <CSSTransition
                unmountOnExit
                className="count"
                classNames="count"
                key={count}
                timeout={{ enter: 400, exit: 400 }}
            >
                <Dot>
                    {count}
                </Dot>
            </CSSTransition>
        </TransitionGroup>
    </AnimationStyles>
);

CartCount.propTypes = {
    count: PropTypes.number.isRequired 
};

export default CartCount;