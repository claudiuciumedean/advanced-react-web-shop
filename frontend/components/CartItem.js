import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import formatMoney from "../lib/formatMoney";

import RemoveFromCart from "./RemoveFromCart";

const CartItemStyles = styled.li`
    display:grid; align-items:center; grid-template-columns:auto 1fr auto; padding:1rem 0; border-bottom:.1rem solid ${props => props.theme.lightgrey};
    img { width:10rem; margin-right:1rem; }
    h3, p { margin:0; }
`;

const CartItem = ({ cartItem }) => {
    if(!cartItem.item) {
        return (
            <CartItemStyles>
                <p>This item has been removed</p>
                <RemoveFromCart id={cartItem.id} />
            </CartItemStyles>
        );
    }

    const { image, title, price } = cartItem.item;
    return (
        <CartItemStyles>
            <img src={image} alt={title}></img>
            <div className="cart-item-details">
                <h3>{title}</h3>
                <p>
                    {formatMoney(price * cartItem.quantity)}
                    -
                    <em>{cartItem.quantity} &times; {formatMoney(price)} each</em>
                </p>
            </div>
            <RemoveFromCart id={cartItem.id} />
        </CartItemStyles>
    )
};


CartItem.propTypes = {
    cartItem: PropTypes.object.isRequired
}

export default CartItem;