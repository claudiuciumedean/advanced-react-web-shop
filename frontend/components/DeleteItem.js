import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";

import { ALL_ITEMS_QUERY } from "./Items";

const DELETE_ITEM_MUTATION = gql`
    mutation DELETE_ITEM_MUTATION($id: ID!) {
        deleteItem(id: $id) {
            id
        }
    }
`;

export default class DeleteItem extends Component {
    render() {
        return (
            <Mutation mutation={DELETE_ITEM_MUTATION} variables={{ id: this.props.id }} update={this.update}>
                {(deleteItem, { err }) => ( <button onClick={() => this.onDelete(deleteItem)}>{this.props.children}</button> )}
            </Mutation>
        );
    }

    onDelete(deleteItem) {
        if(confirm("Are you sure you want to delete this item?")) {
            deleteItem();
        }
    }

    update(cache, payload) {
        const data = cache.readQuery({ query: ALL_ITEMS_QUERY });
        data.items = data.items.filter(item => item.id !== payload.data.deleteItem.id);

        cache.writeQuery({ query: ALL_ITEMS_QUERY, data});
    }
}