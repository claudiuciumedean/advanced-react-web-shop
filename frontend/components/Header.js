import Nav from "./Nav";
import Link from "next/link";
import styled from "styled-components";

const Logo = styled.h1`
    position:relative; transform:skew(7deg); font-size:4rem; margin-left:2rem; z-index:2;
    
`;

const Header = () => (
    <div>
        <div className="bar">
        <h1>
            <Link href="">            
                <a>Sick Fits</a>
            </Link>
        </h1>
        <Nav />
        </div>
        <div className="sub-bar">
        <p>Search</p>
        </div>
        <div>Cart</div>
    </div>
);

export default Header;