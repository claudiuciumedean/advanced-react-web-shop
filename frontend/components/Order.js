import react, { Component } from "react";
import PropTypes from "prop-types";

export default class Order extends Component {
    render() {
        return (
            <p>{this.props.id}</p>
        )
    }

    static propTypes = {
        id: PropTypes.string.isRequired
    }
}