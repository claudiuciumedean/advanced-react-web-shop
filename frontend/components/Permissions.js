import { Query, Mutation } from "react-apollo";
import gql from "graphql-tag";
import PropTypes from "prop-types";

import Error from './ErrorMessage';
import Table from './styles/Table';
import SickButton from './styles/SickButton';

const POSSIBLE_PERMISSIONS = [
    'ADMIN',
    'USER',
    'ITEMCREATE',
    'ITEMUPDATE',
    'ITEMDELETE',
    'PERMISSIONUPDATE',
];

const ALL_USERS_QUERY = gql`
    query {
        users {
            id
            name
            email
            permissions
        }
    }
`;

const UPDATE_PERMISSIONS_MUTATION = gql`
    mutation updatePermissions($permissions: [Permission], $userId: ID!) {
        updatePermissions(permissions: $permissions, userId: $userId) {
            id
            permissions
            name
            email
        } 
    }
`;

const Permissions = props => (
    <Query query={ALL_USERS_QUERY}>
        {({ data, loading, error }) => (
            <div>
                <Error error={error}/>
                <div>
                    <h2>Manage Permissions</h2>
                    <Table>
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Email</th>
                                {POSSIBLE_PERMISSIONS.map((permission, i) => <th key={i}>{permission}</th>)}
                            </tr>
                        </thead>
                        <tbody>
                            {data.users.map((user, i) => <UserPermission user={user} key={i} />)}
                        </tbody>
                    </Table>
                </div>
            </div>
        )}
    </Query>
);

class UserPermission extends React.Component {
    state = {
        permissions: this.props.user.permissions
    };

    render() {
        const user = this.props.user;
        return (
        <Mutation mutation={UPDATE_PERMISSIONS_MUTATION} variables={{permissions: this.state.permissions, userId: this.props.user.id}}>
            {(updatePermissions, { loading, error }) => (
                <>
                    {error && 
                        <tr>
                            <td>
                                <Error error={error}/>
                            </td>
                        </tr>
                    }

                    <tr>
                        <td>{user.name}</td>
                        <td>{user.email}</td>
                        {POSSIBLE_PERMISSIONS.map((permission, i) => (
                            <td key={i}>
                                <label htmlFor={`${user.id}-permission-${permission}`}>
                                    <input type="checkbox" id={`${user.id}-permission-${permission}`} checked={this.state.permissions.includes(permission)} value={permission} onChange={this.handlePermissionChange}/>
                                </label>
                            </td>
                        ))}
                        <td>
                            <SickButton
                                type="button"
                                disabled={loading}
                                onClick={updatePermissions}
                            >
                                Updat{loading ? 'ing' : 'e'}
                            </SickButton>
                        </td>
                    </tr>
                </>
            )}
        </Mutation>
        );
    }

    handlePermissionChange = e => {
        const checkbox = e.target;
        const value = checkbox.value;
        let updatedPermissions = [...this.state.permissions];
        
        if(checkbox.checked) {
            updatedPermissions.push(value);
        } else {
            updatedPermissions = updatedPermissions.filter(permission => permission !== value);
        }

        this.setState({ permissions: updatedPermissions });
    }

    static propTypes = {
        user: PropTypes.shape({
            name: PropTypes.string,
            email: PropTypes.string,
            id: PropTypes.string,
            permissions: PropTypes.array
        }).isRequired
    };
}

export default Permissions;