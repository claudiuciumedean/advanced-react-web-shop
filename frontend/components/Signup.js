import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";

import Form from "./styles/Form";
import Error from "./ErrorMessage";

const SIGNUP_MUTATION = gql`
    mutation SIGNUP_MUTATION($email: String!, $name: String!, $password: String!) {
        signup(email: $email, name: $name, password: $password) {
            id
            email
            name
        }
    }
`;

export default class Signup extends Component {
    state = {
        name: "",
        email: "",
        password: ""
    };

    render() {
        return (
            <Mutation mutation={SIGNUP_MUTATION} variables={this.state}>
                {(signup, {error, loading}) => (          
                <Form method="POST" onSubmit={e => this.onFormSubmit(e, signup)}>
                    <fieldset disabled={loading} aria-busy={loading}>                    
                        <h2>Sign up for an account</h2>
                        <Error error={error}/>

                        <label htmlFor="email">Email</label>
                        <input type="email" name="email" placeholder="email" value={this.state.email} onChange={this.saveToState}/>

                        <label htmlFor="name">Name</label>
                        <input type="text" name="name" placeholder="Name" value={this.state.name} onChange={this.saveToState}/>

                        <label htmlFor="password">Email</label>
                        <input type="password" name="password" placeholder="password" value={this.state.password} onChange={this.saveToState}/>

                        <button type="submit">Sign up!</button>
                    </fieldset>
                </Form>
                )}      
            </Mutation>                      
        );
    }

    saveToState = e => {
        this.setState({ [e.target.name]: e.target.value });
    }

    async onFormSubmit(e, signup) {
        e.preventDefault();
        await signup();
        this.setState({ name: "", email: "", password: "" });
    }
}