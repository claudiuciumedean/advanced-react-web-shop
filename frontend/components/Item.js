import React, { Component } from "react";
import PropTypes from "prop-types";
import Link from "next/link";

import formatMoney from "../lib/formatMoney"; 

import DeleteItem from "./DeleteItem";
import AddToCart from "./AddToCart";

import Title from "./styles/Title";
import ItemStyles from "./styles/ItemStyles";
import PriceTag from "./styles/PriceTag";

export default class Item extends Component {
    render() {
        const { id, image, title, price } = this.props.item;

        return (
            <ItemStyles>
                {image && <img src={image} alt={title}></img>}
                <Title>
                    <Link href={{ pathname: "/item", query: { id: id } }}>
                        <a>{title}</a>
                    </Link>                
                </Title>
                <PriceTag>
                    {formatMoney(price)}
                </PriceTag>
                <div className="buttonList">
                    <Link href={{
                        pathname: "update",
                        query: { id: id }
                    }}>
                        <a>Edit </a>
                    </Link>
                    <AddToCart id={id}/>
                    <DeleteItem id={id}>Delete Item</DeleteItem>
                </div>
            </ItemStyles>
        );
    }

    static propTypes = {
        item: PropTypes.object.isRequired
    };
}