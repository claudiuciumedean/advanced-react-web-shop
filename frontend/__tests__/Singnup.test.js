import { mount } from "enzyme";
import wait from "waait";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import Router from "next/router";

import { fakeUser } from "../lib/testUtils";

import Signup, { SIGNUP_MUTATION } from "../components/Signup";
import { CURRENT_USER_QUERY } from "../components/User";
import { ApolloConsumer } from "react-apollo";

Router.router = {
    push() {}
};
const me = fakeUser();
const { email, name } = me
const mocks = [
    {    
        request: {
            query: SIGNUP_MUTATION,
            variables: { email, name, password:"qwerty" }
        },
        result: {
            data: {
                signup: {
                    __typename: "User",
                    id: "abc123",
                    email,
                    name
                }
            }
        }
    },
    {
        request: {
            query: CURRENT_USER_QUERY
        },
        result: {
            data: { 
                me 
            }
        }
    }
];
const type = (wrapper, name, value) => {
    wrapper.find(`input[name="${name}"]`).simulate("change", {
        target: { name, value }
    });
}

describe("<Signup/>", () => {
    it("renders and mathches snapshot", async () => {
        const wrapper = mount(
            <MockedProvider>
                <Signup/>
            </MockedProvider>
        );

        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("calls the mutation properly", async () => {
        let apolloClient;
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ApolloConsumer>
                    {client => {
                        apolloClient = client;
                        return ( 
                            <Signup/> 
                        );
                    }}
                </ApolloConsumer>
            </MockedProvider>
        );

        await wait();
        wrapper.update();

        type(wrapper, "name", name);
        type(wrapper, "email", email);
        type(wrapper, "password", "qwerty");

        wrapper.update();        
        wrapper.find("form").simulate("submit");

        await wait();

        const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
        expect(user.data.me).toMatchObject(me);
    });
});