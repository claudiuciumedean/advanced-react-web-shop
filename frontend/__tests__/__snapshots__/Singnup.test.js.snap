// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> renders and mathches snapshot 1`] = `
<MockedProvider
  addTypename={true}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "typenameDocumentCache": WeakMap {},
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "typenameDocumentCache": WeakMap {},
            "watches": Array [],
          },
        },
        "version": "2.4.2",
        "watchQuery": [Function],
      }
    }
  >
    <Signup>
      <Mutation
        mutation={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "SIGNUP_MUTATION",
                },
                "operation": "mutation",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "email",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "name",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "password",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "password",
                            },
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "signup",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "email",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NonNullType",
                      "type": Object {
                        "kind": "NamedType",
                        "name": Object {
                          "kind": "Name",
                          "value": "String",
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                    },
                  },
                ],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 217,
              "start": 0,
            },
          }
        }
        refetchQueries={
          Array [
            Object {
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "permissions",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orders",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cart",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "quantity",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "item",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "id",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "price",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "image",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "title",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "description",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 425,
                  "start": 0,
                },
              },
            },
          ]
        }
        variables={
          Object {
            "email": "",
            "name": "",
            "password": "",
          }
        }
      >
        <Form
          method="POST"
          onSubmit={[Function]}
        >
          <form
            className="Form-sc-1xszr8q-0 krRxky"
            method="POST"
            onSubmit={[Function]}
          >
            <fieldset
              aria-busy={false}
              disabled={false}
            >
              <h2>
                Sign up for an account
              </h2>
              <DisplayError
                error={Object {}}
              />
              <label
                htmlFor="email"
              >
                Email
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="email"
                  value=""
                />
              </label>
              <label
                htmlFor="name"
              >
                Name
                <input
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </label>
              <label
                htmlFor="password"
              >
                Password
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
                />
              </label>
              <button
                type="submit"
              >
                Sign up!
              </button>
            </fieldset>
          </form>
        </Form>
      </Mutation>
    </Signup>
  </ApolloProvider>
</MockedProvider>
`;
